sanic_ext/extras/validation/check.py

```diff
@@ -169,7 +177,13 @@ def check_data(model, data, schema, allow_multiple=False, allow_coerce=False):
     except ValueError as e:
         raise TypeError(e)
 
-    return model(**hydration_values)
+    if MSGSPEC and is_msgspec(model):
+        try:
+            return msgspec.from_builtins(hydration_values, model)
```

Comment:
IIUC sanic may validate data from `json`, `query`, and `form` data. For `form` and `query` data using `from_builtins` is the best option since `msgspec` doesn't currently parse querystrings/forms (though there is [an open issue for this](https://github.com/jcrist/msgspec/issues/247)). Since these protocols generally only return strings though, you'll likely want to add a few extra options here to enable coercing more types from string inputs:

```suggestion
            return msgspec.from_builtins(hydration_values, model, str_values=True, str_keys=True)            
```

For `json` data, it'd be nice to parse the JSON payload directly into the requested type, avoiding any unnecessary intermediate allocations. I'm not sure if `sanic`/`sanic-ext` are architected in a way that would enable this currently, but enabling this would be the most performant option.

Reply:
Added this suggestion in https://github.com/sanic-org/sanic-ext/pull/197/commits/7fcd91772188fa3409171277d4a8994d4272c786